src/topp/AssayGeneratorMetabo.cpp

```diff
@@ -433,74 +388,99 @@ class TOPPAssayGeneratorMetabo :
         std::vector<String> subdirs;
-        String out_csifingerid;
-        subdirs = algorithm.callSiriusQProcess(sirius_tmp.getTmpMsFile(),
-                                               sirius_tmp.getTmpOutDir(),
-                                               sirius_executable,
-                                               out_csifingerid,
-                                               decoy_generation);
-  
+        QDirIterator it(sirius_workspace_directory.toQString(), QDir::Dirs | QDir::NoDotAndDotDot, QDirIterator::NoIteratorFlags);
+        while (it.hasNext())
+        {
+          subdirs.emplace_back(it.next());
+        }  
         OPENMS_LOG_DEBUG << subdirs.size() << " spectra were annotated using SIRIUS." << std::endl;
-
         if (subdirs.empty())
         {
-            throw OpenMS::Exception::Postcondition(__FILE__,__LINE__, OPENMS_PRETTY_FUNCTION, "Sirius was executed, but an empty output was generated");
+          decoy_generation = false;
+          throw OpenMS::Exception::Postcondition(__FILE__,__LINE__, OPENMS_PRETTY_FUNCTION, "SIRIUS workspace is empty.");
         }
-
-        // sort vector path list
-        SiriusAdapterAlgorithm::sortSiriusWorkspacePathsByScanIndex(subdirs);
-
-        // extract Sirius/Passatutto FragmentAnnotation and DecoyAnnotation from subdirs
-        // and resolve ambiguous identifications in one file based on the native_id_ids and the SIRIUS IsotopeTree_Score
-        vector<SiriusFragmentAnnotation::SiriusTargetDecoySpectra> annotated_spectra =
-          SiriusFragmentAnnotation::extractAndResolveSiriusAnnotations(subdirs, score_threshold, use_exact_mass);
-
-        // combine compound information (SiriusMSFile) with annotated spectra (SiriusFragmentAnnotation)
-        v_cmp_spec = MetaboTargetedAssay::pairCompoundWithAnnotatedTDSpectraPairs(v_cmpinfo, annotated_spectra);
-
-        // should the sirius workspace be retained
-        if (!sirius_workspace_directory.empty())
+        // sort vector with subdirs by scan index
+        SiriusExportAlgorithm::sortSiriusWorkspacePathsByScanIndex(subdirs);
+        // collect required compound info from each subdir
+        vector<SiriusMSFile::CompoundInfo> v_cmpinfo;
+        for (const auto& subdir : subdirs)
         {
-          // convert path to absolute path
-          QDir sw_dir(sirius_workspace_directory.toQString());
-          sirius_workspace_directory = String(sw_dir.absolutePath());
-
-          // move tmp folder to new location
-          bool copy_status = File::copyDirRecursively(sirius_tmp.getTmpDir().toQString(), sirius_workspace_directory.toQString());
-          if (copy_status)
+          SiriusMSFile::CompoundInfo cmpinfo;
+          // file path for compound.info file
+          String file_path_compound_info = subdir + "/compound.info";
+          // open file and extract information
+          std::ifstream compound_info_file(file_path_compound_info);
+          if (compound_info_file.is_open())
```

Comment:
could we use https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/FORMAT/CsvFile.h
? Mainly to reduce dependency on boost here...